crates/astria-composer/src/executor/bundle_factory/mod.rs

```diff
@@ -153,12 +176,31 @@ impl BundleFactory {
                 })
             }
             Err(SizedBundleError::NotEnoughSpace(seq_action)) => {
-                // if the bundle is full, flush it and start a new one
-                self.finished.push_back(self.curr_bundle.flush());
-                self.curr_bundle
-                    .push(seq_action)
-                    .expect("seq_action should not be larger than max bundle size, this is a bug");
-                Ok(())
+                if self.finished.len() >= self.finished_queue_capacity {
+                    Err(BundleFactoryError::FinishedQueueFull {
+                        curr_bundle_size: self.curr_bundle.curr_size,
+                        finished_queue_capacity: self.finished_queue_capacity,
+                        seq_action,
+                    })
+                } else {
+                    self.finished.push_back(self.curr_bundle.flush());
+                    // if the finished queue is full after the flush mark it as such
+                    if self.finished.len() == self.finished_queue_capacity {
+                        self.full = true;
+                    }
+
+                    self.curr_bundle.try_push(seq_action).expect(
```

Comment:
I still think `expect` here is a little fragile as per [my previous comment](https://github.com/astriaorg/astria/pull/937#discussion_r1562507169).  We could log an `error!` here in case `try_push` returns an error, and just return `BundleFactoryError::SequenceActionTooLarge`

Reply:
the sequence action is already checked in the first `try_push`.

if it's a `SequenceActionTooLarge` error (which happens regardless of the state of `curr_bundle`) it won't reach this branch. getting back a `SequenceActionTooLarge` on the second call can't happen since it was already checked.

since its being pushed to a newly created `SizedBundle` and the `SizedBundle`'s max size is the same as an individual `SequenceAction`'s max size, `NotEnoughSpace` can't happen either.

Reply:
> if it's a `SequenceActionTooLarge` error (which happens regardless of the state of `curr_bundle`) it won't reach this branch.

That's true just now, but if someone in the future swaps the order of the checks to
```rust
if self.curr_size + seq_action_size > self.max_size {
    return Err(SizedBundleError::NotEnoughSpace(seq_action));
}
if seq_action_size > self.max_size {
    return Err(SizedBundleError::SequenceActionTooLarge(seq_action));
}
```
then we could end up panicking here.

I won't belabour the point though.  There's probably little chance that `SizedBundle::try_push` would actually get changed like that (since then we'd never actually hit the `SequenceActionTooLarge` path), and even if it did, we do at least have a test which would expose the panic.

Reply:
If anything, were most likely to get rid of this bundle factory entirely in the near future in favor of using a db